core/src/main/scala/knobs/Parser.scala

```diff
@@ -38,22 +38,22 @@ object ConfigParser {
   }
 
   /** Top-level parser of configuration files */
-  lazy val topLevel: Parser[List[Directive]] = "configuration" |: {
-    directives << skipLWS << realEOF
-  }
+  lazy val topLevel: Parser[List[Directive]] = topLevel(withImport = true)
```

Comment:
It would be possible to separate out the changes that are making the implementation more DRY from the fix to the bug I am trying to address with this PR. The fix to the bug is probably binary compatible (though it would need to be duplicated for both parsers) whereas the refactoring probably needs to break binary compatibility. How import though is it to preserve binary compatibility? I feel like knobs is more of an end-user dependency so it shouldn't matter too much whether this change results in minor or major version bump.